frame/bridge/eth/offchain/src/lib.rs

```diff
@@ -303,24 +318,33 @@ impl<T: Trait> Module<T> {
 
 	/// Build a payload to request the json response or scaled encoded response depence on option
 	fn build_payload(target_number: u64, option: bool) -> Vec<u8> {
-		let mut payload =
-			r#"{"jsonrpc":"2.0","method":"shadow_getEthHeaderWithProofByNumber","params":{"block_num":"#
-				.as_bytes()
-				.to_vec();
-		payload.append(&mut base_n_bytes_unchecked(target_number, 10));
-		payload.append(&mut r#","transcation":false"#.as_bytes().to_vec());
+		let header_part: &[u8] = br#"{"jsonrpc":"2.0","method":"shadow_getEthHeaderWithProofByNumber","params":{"block_num":"#;
+		let number_part: &[u8] = &base_n_bytes_unchecked(target_number, 10)[..];
+		let transcation_part: &[u8] = br#","transcation":false"#;
+		let option_part: &[u8] = br#","options":{"format":"json"}"#;
+		let tail: &[u8] = br#"},"id":1}"#;
+
 		if option {
-			payload.append(&mut r#","options":{"format":"json"}"#.as_bytes().to_vec());
+			[
+				header_part,
+				number_part,
+				transcation_part,
+				option_part,
+				tail,
+			]
+			.concat()
+		} else {
+			[header_part, number_part, transcation_part, tail].concat()
 		}
-		payload.append(&mut r#"},"id":1}"#.as_bytes().to_vec());
-		payload
 	}
 
 	/// Extract the proof no mater the response is scale encoded format or json format
 	fn extract_proof(r: &mut Vec<u8>, option: bool) {
+		// Note the size is checked in `validate_response`, so there is no check here
+
 		let (hint, left_offset, right_offset) = if option { (125, 11, 5) } else { (44, 12, 3) };
-		let mut pr = 47;
-		for i in 47..r.len() {
+		let mut pr = 1355;
```

Comment:
I think the numbers here are quite magic, need some comments on it. Or better to use robust utils/libraries instead.

Reply:
Fixed, thanks for review.